lib/Parse/ParseDecl.cpp

```diff
@@ -4909,6 +4910,104 @@ ParserStatus Parser::parseTypeAttribute(TypeAttributes &Attributes,
   return makeParserSuccess();
 }
 
+static llvm::Optional<LifetimeDependenceKind>
+getLifetimeDependenceKind(const Token &T) {
+  if (T.isContextualKeyword("_copy")) {
+    return LifetimeDependenceKind::Copy;
+  }
+  if (T.isContextualKeyword("_consume")) {
+    return LifetimeDependenceKind::Consume;
+  }
+  if (T.isContextualKeyword("_borrow")) {
+    return LifetimeDependenceKind::Borrow;
+  }
+  if (T.isContextualKeyword("_mutate")) {
+    return LifetimeDependenceKind::Mutate;
+  }
+  return llvm::None;
+}
+
+ParserStatus Parser::parseLifetimeDependenceSpecifiers(
+    SmallVectorImpl<LifetimeDependenceSpecifier> &specifierList) {
+  ParserStatus status;
+  // TODO: Add fixits for diagnostics in this function.
+  do {
+    auto lifetimeDependenceKind = getLifetimeDependenceKind(Tok);
+    if (!lifetimeDependenceKind.has_value()) {
+      break;
+    }
+    // consume the lifetime dependence kind
+    consumeToken();
```

Comment:
Not remembering the location of kind token makes `getStartLoc()` incorrect, but I don't think it's serious for now. `applyAttributesToType()` changes the order of specifiers anyway. We should fix them, but not necessary in this PR.

Also, representing single `_copy(x, y)` as separate `LifetimeDependenceSpecifier` make it impossible to restore the textual representation of the specifier, but I think that's also not serious for now.